### ISSUE TITLE
Export parseModule and generateCode in a different module that doesn't rely on node api

### ISSUE THREAD

**Issue by paoloricciuti** [unjs/magicast#72]
### Describe the feature

Currently `parseModule` and `generateCode` are exported from the code module that import `node:fs` this made them impossible to use in a browser environment when actually they are not really dependent on node apis.

Maybe we can consider export them from another module that does not depends on node api to allow their usage even in the browser.

### Additional information

- [X] Would you be willing to help implement this feature?

**Comment by antfu** [comment]
Yeah I think it makes sense to have a submodule for that. PR Welcome :)

**Comment by paoloricciuti** [comment]
> Yeah I think it makes sense to have a submodule for that. PR Welcome :)

Should I still re-export them from the current module to avoid breaking changes?

**Comment by antfu** [comment]
Yes. I think we could have `magicast/browser` or `magicast/core` for that.

**Comment by Hugos68** [comment]
@antfu Do you want `magicast/browser` to also re export all other things like types, helpers etc, or have `magicast/browser` only export `parseModule` and `generateCode`. I'm in need of magicast in a browser env. Creating a PR as we speak.
